hoot-core/src/main/cpp/hoot/core/elements/ElementComparer.cpp

```diff
@@ -179,13 +191,39 @@ bool ElementComparer::_compareWay(const std::shared_ptr<const Element>& re,
 
   if (rw->getNodeIds().size() != w->getNodeIds().size())
   {
+    LOG_TRACE(
+      "Ways " << rw->getElementId() << " and " << w->getElementId() <<
+      " failed comparison on way node count: " << rw->getNodeIds().size() << " and " <<
+      w->getNodeIds().size());
     return false;
   }
-  for (size_t i = 0; i < rw->getNodeIds().size(); ++i)
+
+  if (!_ignoreElementId)
   {
-    if (rw->getNodeIds()[i] != w->getNodeIds()[i])
+    for (size_t i = 0; i < rw->getNodeIds().size(); ++i)
     {
-      return false;
+      if (rw->getNodeIds()[i] != w->getNodeIds()[i])
+      {
+        LOG_TRACE(
+          "Ways " << rw->getElementId() << " and " << w->getElementId() <<
+          " failed comparison on way nodes: " << rw->getNodeIds() << " and " << w->getNodeIds());
+        return false;
+      }
+    }
+  }
+  else
+  {
+    for (size_t i = 0; i < rw->getNodeIds().size(); ++i)
```

Comment:
We're ignoring element IDs during comparison here. Not only can the way IDs differ here, but their way node IDs can as well. So, we need to compare each way node between the ways to each other. The resulting node hash calc seems like it could be a little expensive for this, but I didn't notice any performance differences when running the regression tests. The way tag size and way node size checks should that come first will prevent a lot of these comparisons from having to be made.